### ISSUE TITLE
Use system emoji

### ISSUE THREAD

**Issue by skjnldsv** [nextcloud/talk-android#543]
I use specific choose emoji on my android phone and I like to have a constant style when I use them.

Please use the built-in emoji font and not use any custom one :)

**Comment by mario** [comment]
(on vacation, but I see you added developing so I have to react)...

It was previously agreed that we use Twemoji (Twitter emojis) everywhere for standardization as evidenced here: https://github.com/nextcloud/server/issues/10293

**Comment by skjnldsv** [comment]
@mario get back to your vacations and answer when you're back ;)

Pinging @jancborchardt

**Comment by jancborchardt** [comment]
Yep, we standardized on Twemoji. Other platforms do it similarly, and this way everyone seeing the emojis sees the same thing.

**Comment by skjnldsv** [comment]
@jancborchardt well, same goes for the enforced font.
Please reconsider using the system emoji. Creating a new standard for us when there is already an existing one is not a good idea imho :)

| Using system | Overriding defaults |
|:---------:|:------:|
|Signal|Whatsapp|
|Textra|Messenger|
|Android stock messaging|-|
|Instagram|-|
|Telegram|-|
|Snapchat|-|

**Comment by jancborchardt** [comment]
We don't enforce the font anymore, yes. :)

Signal does override the default btw, they use iOS emojis on Android as well as desktop when you pick them via their emoji selector in the text input. (You can get to the Android emojis through the keyboard, but the posted ones will be iOS style.)

So in the table, double-check that the posted emojis are actually the same. And what happens for the desktop apps.

**Comment by jancborchardt** [comment]
Just checked (play store, don't use the apps so might be wrong):
- [Instagram](https://play.google.com/store/apps/details?id=com.instagram.android&hl=en) in one of their screenshots shows Android emojis, in the last one it's iOS
- [Snapchat](https://play.google.com/store/apps/details?id=com.snapchat.android&hl=en_US) shows iOS emojis in the screenshot.
- [Telegram](https://play.google.com/store/apps/details?id=org.telegram.messenger) looks like it's using iOS emoji too on Android and desktop?

So I guess it's rather a question if we should / can use iOS emojis across platforms?

**Comment by skjnldsv** [comment]
Ah, I might have mischecked then.
I changed the system default on my phone with the iOS ones, so seeing telegram with iOS styling made my brain connect the two :)
Isn't it weird that they use it? Is it really allowed to them?

**Comment by jancborchardt** [comment]
(And Android stock messaging and Textra don't really count as they are not cross-platform. Textra even has a selector for the Emoji style, including Twemoji and iOS. ;D)

**Comment by skjnldsv** [comment]
Okay, so Instagram uses android default. Their screenshot is not representative of what you get.
Telegram uses iOS style
Snapchat uses default android

**Comment by skjnldsv** [comment]
> Textra even has a selector for the Emoji style, including Twemoji and iOS. ;D)

Exactly. You can choose the default or override it in the app.
Default is the app, you need to install additionnal extension to get the other emoji styles :)

**Comment by jancborchardt** [comment]
> Isn't it weird that they use it? Is it really allowed to them?

Yeah, that's what I was thinking too :/ not sure about it. But that's also a part of why we use Twemoji, cause we need to ship smth with the server and that should have an open license.

**Comment by skjnldsv** [comment]
So, in the apps I tested, we're now at almost 50/50. (Table updated)
It does not seems to have *a* way of doing it. So I'd still argue my case by stating it's frustrating to have an app mess with your system settings. Same argument I had for the enforced system font. :)

**Comment by skjnldsv** [comment]
Tested a few more popular apps (which I never used):
- Line, 11.4M dls: Default system
- Viber, 12.8M dls: USing another emoji, don't know which one :thinking: 
- Wechat, 5.6M dls: iOS style

**Comment by jancborchardt** [comment]
So regarding "which emojis does the app use on Android?" we have this so far:
- Android/Google style: Instagram, Snapchat, Line
- iOS style: Signal, Telegram, Wechat
- Custom (all different ones): Facebook, Messenger, WhatsApp, Viber

Messenger lite and Facebook lite don’t really count as both Facebook and Messenger don’t use Android, and of course a lite version wouldn’t ship anything out of the bare bones.
Also Textra and Android stock messaging don’t really count as they are Android only.
And lastly remember Instagram is owned by Facebook and adjusting to it more and more, so it’s highly likely that it will go the way of Facebook, Messenger and WhatsApp (which all use separate custom styles though, see [Emojipedia](https://emojipedia.org/face-with-tears-of-joy/) for reference).

I’m not really hard on having a specific style on Android, but the point is that we need to check cross-platform:
- We need something for the web interface (and future possible desktop app) anyway. There’s no standard here.
- What happens if you use a branded Android like from Samsung? How do the other apps do it?
- What do these apps do on iOS, and on desktop? (Brief test on iOS: Basically all apps except Viber use iOS emoji.)

**TL;DR the de facto standard (or most used) seems to be iOS emoji, and if we use anything else than Twemoji in a cross-platform way, then it should probably be the iOS style.**

**Comment by skjnldsv** [comment]
> Messenger lite and Facebook lite don’t really count as both Facebook and Messenger don’t use Android, and of course a lite version wouldn’t ship anything out of the bare bones.

That is why I added both of the versions. I'd say they count as well. They're an interesting option.

> And lastly remember Instagram is owned by Facebook and adjusting to it more and more, so it’s highly likely that it will go the way of Facebook, Messenger and WhatsApp (which all use separate custom styles though, see [Emojipedia](https://emojipedia.org/face-with-tears-of-joy/) for reference). 

Well, let's not draw conclusions for the sake of going your way :grin: 

> I’m not really hard on having a specific style on Android, but the point is that we need to check cross-platform:
> * We need something for the web interface (and future possible desktop app) anyway. There’s no standard here.

Why so? Why would our standard override the android's one? I actually prefer having a unity on my phone than across an app on different platforms :)

> * What happens if you use a branded Android like from Samsung? How do the other apps do it?

Same apply, samsung's style just use a different emoji font. So It will be the same as android's default where apps uses it


> * What do these apps do on iOS, and on desktop? (Brief test on iOS: Basically all apps except Viber use iOS emoji.)

Yep, like we should do, use android's default. Even Twitter (which we use the eomji) uses android's default and not their own on the official twitter app :)

**Comment by jancborchardt** [comment]
(Just for the record as we talked about this via chat too:)

Often it happens to me that people send me emojis and they are not displayed because it’s a hot new emoji and not supported yet. Signal actually has a page on that: https://support.signal.org/hc/en-us/articles/360017561992-Why-are-the-emoji-on-Signal-Android-different-from-the-other-apps-on-my-phone-

> Signal Android includes built-in emoji functionality for consistency between platforms and to support older Android devices whose set of emoji may be outdated or incomplete. However, if you prefer to use the emoji that shipped with your phone, here's how to switch:
> 1. Open Signal, tap your profile profile_avatar.png > Chats and media.
> 2. Under Chats, enable Use system emoji.

**Comment by jancborchardt** [comment]
3 possible solutions (could be more):
- We keep Twemoji: Looks strange on iOS and Android, but is uniform across server and possible desktop app, and is openly licensed
- We use iOS style: Looks strange on Android, but a third of surveyed apps use it so not _so_ strange. But it’s not openly licensed so we can probably not do this anyway.
- We use system default: Looks consistent on the device, but will break when your phone is just slightly older and not show certain emojis. This is why the majority of examples listed above _don’t_ use the system default, as also explained by the Signal quote.

**Comment by mario** [comment]
Using the built-in emoji set is very weird because it will not work as expected on many phones. On the other hand, I recognize that you might dislike Twitter set. For that reason (and because it solves a few other issues we've been having) I'll switch to the default Google emoji font (bundled) which will look the same across all the Android devices.

Not perfect, but better than status quo that we have now.

**Comment by skjnldsv** [comment]
> default Google emoji font (bundled)

so we're shipping the latest emoji fonts?

**Comment by mario** [comment]
Yes @skjnldsv

**Comment by skjnldsv** [comment]
Could we send a mail to Apple asking for permission to ship theirs? :p
They're far more advanced and good looking than the cartoon-style google ones :see_no_evil:

**Comment by skjnldsv** [comment]
![Capture d’écran_2019-07-22_13-07-49](https://user-images.githubusercontent.com/14975046/61628128-b8a39e00-ac81-11e9-94e7-58efd749a3d9.png)
